src/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeReader.cs

```diff
@@ -308,7 +310,7 @@ private void AllocateReadTail()
                 _readHead = AllocateSegment();
                 _readTail = _readHead;
             }
-            else if (_readTail.WritableBytes < _minimumReadThreshold)
+            else if (_readTail!.WritableBytes < _minimumReadThreshold)
```

Comment:
You might instead do:
```C#
else
{
    Debug.Assert(_readTail != null);
    if (_readTail.WritableBytes < _minimumReadThreshold)
    {
        ...
    }
}
```
That way, there's consistency between the two branches, each asserting at the beginning the state of `_readTail`.